test/server/server_fuzz_test.cc

```diff
@@ -103,11 +103,32 @@ bool validateLbSubsetConfig(const envoy::config::bootstrap::v3::Bootstrap& input
   return true;
 }
 
+bool validateUpstreamConfig(const envoy::config::bootstrap::v3::Bootstrap& input) {
+  for (auto const& cluster : input.static_resources().clusters()) {
+    if (Envoy::Config::Utility::getFactory<Envoy::Router::GenericConnPoolFactory>(
+            cluster.upstream_config()) == nullptr) {
+      ENVOY_LOG_MISC(debug, "upstream_config: typed config {} invalid",
+                     cluster.upstream_config().DebugString());
+      return false;
+    }
+  }
+  return true;
+}
+
 DEFINE_PROTO_FUZZER(const envoy::config::bootstrap::v3::Bootstrap& input) {
+  try {
+    TestUtility::validate(input);
```

Comment:
The CI for target `bazel.compile_time_options` activates `--define "deprecated_features=disabled"`. The `TestUtility::validate(input)` now no longer flags deprecated config-options as such, but errors out. The corpi for the `server_fuzz_test` have several deprecated config members in them. I see three ways to resolve this:
1. Remove the `TestUtility::validate(input);` including the try-catch, or
2. Remove the deprecated config options from the corpi, or
3. Catch the exception thrown for the deprecation and handle it.

Option 3. might seem to be good, but note, that it will catch only the first issue and then no longer validate the config.
Which way shall I go?

Reply:
from: https://github.com/envoyproxy/envoy/blob/583ae63b1525ea231900ae7da53c2c1cdc93c080/source/common/protobuf/utility.cc#L130-L133

It seems just like we upgrade tests to not use those deprecated fields, we might also want to do that for fuzzers.

I think the validation is very useful for constraining the fuzzer.

Reply:
+1 to @KBaichoo. If it's not too hard to "upgrade" the corpus files, let's do that (so we can make sure the test still passes, and let the fuzzer seed corpus be more meaningful).
But let me know if this is a non-trivial change, and I'll take a deeper look.

Reply:
It seems many of the corpus use `envoy.config.cluster.v3.Cluster.common_http_protocol_options` which is now deprecated. According to the comment of [this field](https://github.com/envoyproxy/envoy/blob/7dc822ec4f8540897cedad1c7d772cb14be5e1cc/api/envoy/config/cluster/v3/cluster.proto#L841), it was deprecated (in [PR 14079](https://github.com/envoyproxy/envoy/pull/14079/files)) and it can be set by using `typed_extension_protocol_options`.

Here's an example that I've seen:
* Before:
```
    common_http_protocol_options:
      idle_timeout: 0s
```
* After:
```
    typed_extension_protocol_options:
      envoy.extensions.upstreams.http.v3.HttpProtocolOptions:
        "@type": type.googleapis.com/envoy.extensions.upstreams.http.v3.HttpProtocolOptions
        common_http_protocol_options:
          idle_timeout: 0s
```

Reply:
I tried to fix the corpi to my best understanding. There was a lot to do. I hope to have caught every issue. For the options I've removed the fuzzer will hopefully select in the future again. 

Btw, we need to tell the fuzzer to stop using deprecated options. Do you know where this is configured `--define deprecated_features=disabled`, or shall I investigate?

Reply:
Thanks for cleaning this up!
If you can let the fuzzer run for a couple of hours, and it won't find new crashes, then we can move this PR to the open repo (just to make sure we don't make it too easy to find new bugs).

I think that for now we'll want to keep the fuzzer using deprecated (but still available) functionality, as changing all the configs will be a non-trivial work and will take a longer time.